src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensEditsHandler.cs

```diff
@@ -0,0 +1,377 @@
+            using var _ = ArrayBuilder<SemanticToken>.GetInstance(out var fullTokens);
+            for (var i = 0; i < tokens.Length; i += 5)
+            {
+                fullTokens.Add(new SemanticToken(tokens[i], tokens[i + 1], tokens[i + 2], tokens[i + 3], tokens[i + 4]));
+            }
+
+            return fullTokens.ToArray();
+        }
+
+        internal static LSP.SemanticTokensEdit GenerateEdit(int start, int deleteCount, int[] data)
+            => new LSP.SemanticTokensEdit
+            {
+                Start = start,
+                DeleteCount = deleteCount,
+                Data = data
+            };
+
+        private sealed class LongestCommonSemanticTokensSubsequence : LongestCommonSubsequence<SemanticToken[]>
+        {
+            private static readonly LongestCommonSemanticTokensSubsequence s_instance = new LongestCommonSemanticTokensSubsequence();
+
+            protected override bool ItemsEqual(
+                SemanticToken[] oldSemanticTokens, int oldIndex,
+                SemanticToken[] newSemanticTokens, int newIndex)
+                => oldSemanticTokens[oldIndex].Equals(newSemanticTokens[newIndex]);
+
+            public static IEnumerable<SequenceEdit> GetEdits(
+                SemanticToken[] oldSemanticTokens, SemanticToken[] newSemanticTokens)
+                => s_instance.GetEdits(oldSemanticTokens, oldSemanticTokens.Length, newSemanticTokens, newSemanticTokens.Length);
+        }
+
+        /// <summary>
+        /// Stores the values that make up the LSP representation of an individual semantic token.
+        /// </summary>
+        private readonly struct SemanticToken
+        {
+            private readonly int _deltaLine;
+            private readonly int _deltaStartCharacter;
+            private readonly int _length;
+            private readonly int _tokenType;
+            private readonly int _tokenModifiers;
+
+            public SemanticToken(int deltaLine, int deltaStartCharacter, int length, int tokenType, int tokenModifiers)
+            {
+                _deltaLine = deltaLine;
+                _deltaStartCharacter = deltaStartCharacter;
+                _length = length;
+                _tokenType = tokenType;
+                _tokenModifiers = tokenModifiers;
+            }
+
+            public int[] ToArray()
+            {
+                return new int[] { _deltaLine, _deltaStartCharacter, _length, _tokenType, _tokenModifiers };
+            }
+
+            public override bool Equals(object? obj)
+            {
+                return obj is SemanticToken token &&
+                       _deltaLine == token._deltaLine &&
+                       _deltaStartCharacter == token._deltaStartCharacter &&
+                       _length == token._length &&
+                       _tokenType == token._tokenType &&
+                       _tokenModifiers == token._tokenModifiers;
+            }
+
+            public override int GetHashCode()
```

Comment:
remove.  you never hash these.  and move .Equals(object) and replace with .Equals(SemanticToken)